builder/bundleset_test.go

```diff
@@ -0,0 +1,195 @@
+			}
+			if !samePackages {
+				msg += fmt.Sprintf("\nPARSED PACKAGES (%d):\n%s\nEXPECTED PACKAGES (%d):\n%s", len(packages), packages, len(tt.ExpectedPackages), tt.ExpectedPackages)
+			}
+			t.Errorf(msg)
+			continue
+		}
+	}
+}
+
+func TestParseBundleSet(t *testing.T) {
+	type FilesMap map[string]string
+	type CountsMap map[string]int
+
+	// Repurpose empty CountsMap as error to make the test entries less verbose, since we can avoid
+	// putting the attribute names in the literal map.
+	Error := CountsMap{}
+
+	// Replace spaces with new lines.
+	Lines := func(s string) string {
+		return strings.Replace(s, " ", "\n", -1)
+	}
+
+	tests := []struct {
+		Name  string
+		Files FilesMap
+		// TODO: Replace this with the actual map.
+		ExpectedAllPackageCounts CountsMap
+	}{
+		{
+			"simple include",
+			FilesMap{
+				"a": Lines("A1 A2"),
+				"b": "include(a)",
+			},
+			CountsMap{
+				"a": 2,
+				"b": 2,
+			},
+		},
+		{
+			"redundant includes",
+			FilesMap{
+				"a": Lines("A1 A2 A3 A4"),
+				"b": Lines("include(a) B1 B2 B3 B4"),
+				"c": Lines("include(b) C1 C2 C3 C4"),
+				"d": Lines("include(a) include(b) include(c) A1 B1 C1 D1"),
+			},
+			CountsMap{
+				"a": 4,
+				"b": 8,
+				"c": 12,
+				"d": 13,
+			},
+		},
+
+		{
+			"all packages don't have duplicates",
+			FilesMap{
+				"a": Lines("A A A A"),
+				"b": Lines("include(a) A A A A"),
+				"c": Lines("include(b) A A A A include(a)"),
+				"d": Lines("A"),
+				"e": Lines("include(a) include(d) E"),
+			},
+			CountsMap{
+				"a": 1,
+				"b": 1,
+				"c": 1,
+				"d": 1,
+				"e": 2,
+			},
+		},
+
+		{"cyclic error two bundles",
+			FilesMap{"a": "include(b)", "b": "include(a)"}, Error},
+
+		{"cyclic error three bundles",
+			FilesMap{"a": "include(b)", "b": "include(c)", "c": "include(a)"}, Error},
+
+		{"bundle not available",
+			FilesMap{"a": "include(c)"}, Error},
+
+		{"bundle not available 2",
+			FilesMap{"a": "include(b)", "b": "include(c)"}, Error},
+	}
+
+	testDir, err := ioutil.TempDir("", "bundleset-test-")
+	if err != nil {
+		t.Fatalf("couldn't create temporary directory to write test cases: %s", err)
+	}
+	defer func() {
+		_ = os.RemoveAll(testDir)
+	}()
+
+	for i, tt := range tests {
+		dir := filepath.Join(testDir, fmt.Sprint(i))
+		err = os.Mkdir(dir, 0700)
+		if err != nil {
+			t.Fatalf("couldn't create temporary directory to write test case: %s", err)
+		}
+
+		bundleFiles := make([]string, 0, len(tt.Files))
+		for name, contents := range tt.Files {
+			bundleFile := filepath.Join(dir, name)
+			err = ioutil.WriteFile(bundleFile, []byte(contents), 0600)
+			if err != nil {
+				t.Fatalf("couldn't create temporary file for test case: %s", err)
+			}
+			bundleFiles = append(bundleFiles, bundleFile)
+		}
+
+		expectedError := (len(tt.ExpectedAllPackageCounts) == 0)
+
+		var set *bundleSet
+		set, err = newBundleSet(bundleFiles)
+		if err != nil {
```

Comment:
You could simplify this as I mentioned in an above comment.

Reply:
Done, also renaming to shouldFail.